raiden/app.py

```diff
@@ -59,11 +59,11 @@ class App:  # pylint: disable=too-few-public-methods
         'settle_timeout': DEFAULT_SETTLE_TIMEOUT,
         'database_path': '',
         'msg_timeout': 100.0,
-        'protocol': {
-            'retry_interval': DEFAULT_PROTOCOL_RETRY_INTERVAL,
-            'retries_before_backoff': DEFAULT_PROTOCOL_RETRIES_BEFORE_BACKOFF,
-            'throttle_capacity': DEFAULT_PROTOCOL_THROTTLE_CAPACITY,
-            'throttle_fill_rate': DEFAULT_PROTOCOL_THROTTLE_FILL_RATE,
+        'transport': {
```

Comment:
@ulope @andrevmatos I'd like to move the `matrix` key inside this section, and to group the current content inside a key named `udp`, comments?

Reply:
@hackaugusto hmm, that would look like we're going to support both transports. I'd prefer to use different keys and just put both under `transport`. If the keys doesn't conflict, no need to separate them, they'd imply _the default transport_ anyway.

Reply:
>  I'd prefer to use different keys and just put both under transport

That is what I meant, e.g.:

```
config = {
  'transport': {
     'udp': { ... },
     'matrix': { ... },
  }
}
```

Right now the transport configuration is sort of spread out.

Reply:
@hackaugusto yes, I understood. What I suggested is like:
```
config = {
  'transport': {
    'udpparam1': ...,
    'udpparam2': ...,
    'sharedparam1': ...,
    'matrixparam1': ...
  }
}
```
No udp nor matrix prefixes required, just to exemplify. The idea is that the transport, whichever it be, look for its parameter in the `transport` config. No need to split in separate subkeys/namespaces, each transport will look for the keys it cares, and ignore the others, and it means we only support one transport anyway.

Reply:
yeah, I dislike your approach It's unclearer since config keys are mixed and it assumes that we will never have more than one transport

Reply:
> it assumes that we will never have more than one transport

That's the idea. We already talked about how multiple transports shouldn't coexist. You can merge `matrix` key in the transport, but I'd rather not have it in a subkey.

Reply:
The advantage of grouping the parameters by transport type is that it's immediately clear which is used by which type. Parameters that affect all transport types could be placed under a `common` (or whatever) key. 

Since we've already talked about that we *may* switch to a "proper" P2P transport at some point in the future this seems like the cleanest and most future proof solution.